colorama/ansitowin32.py

```diff
@@ -21,6 +21,35 @@ def is_a_tty(stream):
     return hasattr(stream, 'isatty') and stream.isatty()
 
 
+def check_windows_ansi_support():
+    try:
+        import winreg
+    except ImportError:
+        try:
+            import _winreg  # on Python 2
+        except ImportError:
+            return False
+        else:
+            winreg = _winreg
+
+    # Check the registry for release ID
+    key = r"SOFTWARE\Microsoft\Windows NT\CurrentVersion"
+    val = r"ReleaseID"
+    key = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, key)
+    releaseId = int(winreg.QueryValueEx(key,val)[0])
+    winreg.CloseKey(key)
+
+    # Windows 10 supports ANSI cmd since release 1511
+    if releaseId >= 1511:
```

Comment:
You probably need to check that this is Windows 10 too, I'm not sure ReleaseID isn't used in previous versions of Windows. l Windows since 8.1 has a nasty habit of lying about it's version though. I think `platform.win32_ver()` does work around that but be sure to check (`sys.getwindowsversion` doesn't workaround this at least on Python 2.7.13 so you get the version of Windows 8).